pkg/nuclide-clang-rpc/lib/ClangFlagsManager.js

```diff
@@ -266,6 +304,34 @@ export default class ClangFlagsManager {
     return null;
   }
 
+  async _loadProjectCompilerFlags(flagsFile: string): Promise<?ClangProjectFlags> {
+    if (this._clangProjectFlags != null) {
```

Comment:
As mentioned above, you need to look this up by `flagsFile`. And you need to cache the promise- see the `getFlagsForSrc` cached vs impl pattern above